### PR TITLE
[Merged by Bors] - feat(AddChar): more basic lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
Also unsimp `AddChar.div_apply`/`AddChar.sub_apply` because it doesn't do the right thing in my application.

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/AddChar.lean [hunk 1]
```diff
@@ -408,6 +425,16 @@ lemma injective_iff {ψ : AddChar A M} : Injective ψ ↔ ∀ ⦃x⦄, ψ x = 1
 
 end fromAddGrouptoDivisionCommMonoid
 
+section fromAddCommGrouptoDivisionCommMonoid
+variable {G M : Type*} [AddCommGroup G] [DivisionCommMonoid M]
```
Comment (b-mehta) [inline Mathlib/Algebra/Group/AddChar.lean:429]: It looks like this section has exactly the same assumptions as the previous

File: Mathlib/Algebra/Group/AddChar.lean [hunk 2]
```diff
@@ -408,6 +425,16 @@ lemma injective_iff {ψ : AddChar A M} : Injective ψ ↔ ∀ ⦃x⦄, ψ x = 1
...
+
+@[simp] lemma zsmul_apply (n : ℤ) (ψ : AddChar G M) (a : G) : (n • ψ) a = ψ a ^ n := by
```
Comment (b-mehta) [inline Mathlib/Algebra/Group/AddChar.lean:431]: Let's add the multiplicative version of this too

File: Mathlib/Algebra/Group/AddChar.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Group/AddChar.lean:429]: Oh, the above section is misnamed

File: Mathlib/Algebra/Group/AddChar.lean [hunk 3]
```diff
@@ -291,6 +303,11 @@ lemma mul_eq_add (ψ χ : AddChar A M) : ψ * χ = ψ + χ := rfl
 lemma pow_eq_nsmul (ψ : AddChar A M) (n : ℕ) : ψ ^ n = n • ψ := rfl
 lemma prod_eq_sum (s : Finset ι) (ψ : ι → AddChar A M) : ∏ i in s, ψ i = ∑ i in s, ψ i := rfl
 
+@[simp] lemma toMonoidHomEquiv_add (ψ φ : AddChar A M) :
+    toMonoidHomEquiv (ψ + φ) = toMonoidHomEquiv ψ * toMonoidHomEquiv φ := rfl
+@[simp] lemma toMonoidHomEquiv_symm_mul (ψ φ : Multiplicative A →* M) :
+  toMonoidHomEquiv.symm (ψ * φ) = toMonoidHomEquiv.symm ψ + toMonoidHomEquiv.symm φ := rfl
```
Comment (b-mehta) [inline Mathlib/Algebra/Group/AddChar.lean:309]:
```suggestion
    toMonoidHomEquiv.symm (ψ * φ) = toMonoidHomEquiv.symm ψ + toMonoidHomEquiv.symm φ := rfl
```

File: Mathlib/Algebra/Group/AddChar.lean [hunk 4]
```diff
@@ -400,13 +416,20 @@ lemma div_apply' (ψ χ : AddChar A M) (a : A) : (ψ / χ) a = ψ a / χ a := by
 lemma sub_apply' (ψ χ : AddChar A M) (a : A) : (ψ - χ) a = ψ a / χ a := by
   rw [sub_apply, map_neg_eq_inv, div_eq_mul_inv]
 
+@[simp] lemma zsmul_apply (n : ℤ) (ψ : AddChar A M) (a : A) : (n • ψ) a = ψ a ^ n := by
+  cases n
+  · simp
+  · simp [-neg_apply, neg_apply']
```
Comment (b-mehta) [inline Mathlib/Algebra/Group/AddChar.lean:422]: could make this a cute one-line proof if you want

**Review by b-mehta — APPROVED** [review]
bors d+

File: Mathlib/Algebra/Group/AddChar.lean (hunk 4 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Group/AddChar.lean:422]: Well spotted

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**